scripts/scripts/security.groovy

```diff
@@ -0,0 +1,37 @@
+import hudson.security.*
+import jenkins.model.Jenkins
+
+
+def authStrategy = config.authStrategy
+println "Authorization strategy is: ${authStrategy}"
+
+def instance = Jenkins.getInstance()
+
+switch (authStrategy) {
+    case 'Logged-in users can do anything':
+        def allowAnonymousRead = config?.allowAnonymousRead ?: false
+        setDefaultStrategy(instance, allowAnonymousRead)
+        break
+    case 'Matrix-based security':
+        setMatrixStrategy(instance, config.matrix)
+        break
+    default:
+        throw new RuntimeException("Authorization strategy ${authStrategy} not supported.")
+
+}
+
+def setMatrixStrategy(instance, matrix) {
+    def strategy = new GlobalMatrixAuthorizationStrategy()
+    matrix.each { role, user ->
```

Comment:
it's not necessarily a user, can be a group,
the plugin seems to call this a `sid` security identifier, not sure if you would like to rename it to something else?